app/main/posts/views/filters/active-search-filters.directive.js

```diff
@@ -46,11 +83,36 @@ function ActiveSearchFilters($translate, $filter, PostFilters, _, FilterTransfor
             return value;
         }
 
-        function removeFilter(filterKey, value, $event) {
+        function removeFilter(filterKey, value, savedSearch, $event) {
             $event.preventDefault();
             $event.stopPropagation();
-            PostFilters.clearFilter(filterKey, value);
+            if (savedSearch) {
+                savedSearch.filter = PostFilters.clearFilterFromArray(filterKey, value, savedSearch.filter);
+                PostFilters.clearFilter(filterKey, value);
+                PostFilters.setFilter('saved_search', savedSearch);
```

Comment:
why do we need to set saved_search as a filter again here?

Reply:
We don't 👍